DevIncubator-Autopark/Collections.cs

```diff
@@ -0,0 +1,169 @@
+﻿namespace DevIncubatorAutopark
+{
+    internal class Collections
+    {
+        public List<VehicleType> VehicleTypes { get; } = new List<VehicleType>();
+        public List<Vehicle> Vehicles { get; } = new List<Vehicle>();
+
+        public Collections()
+        {
+
+        }
+        public Collections(string rentsFileName, string vehiclesFileName, string vehiclesTypesFileName)
+        {
+            VehicleTypes = ParseVehicleTypes(vehiclesTypesFileName);
+            Vehicles = ParseVehicles(vehiclesFileName);
+            LoadRents(rentsFileName);
+        }
+
+        private VehicleType GetVehicleTypeById(int id)
+        {
+            foreach (var vehicleType in VehicleTypes)
+            {
+                if (vehicleType.Id == id)
+                {
+                    return vehicleType;
+                }
+            }
+            return null;
+        }
+
+        private void CreateRent(IReadOnlyList<string> fields)
+        {
+            var vehicleId = int.Parse(fields[0]);
+            var rentTime = DateTime.Parse(fields[1]);
+            var rentPrice = double.Parse(fields[2]);
+
+            foreach (var vehicle in Vehicles)
+            {
+                if (vehicle.Id == vehicleId)
+                {
+                    vehicle.Rents.Add(new Rent(rentTime, rentPrice));
+                }
+            }
+        }
+
+        private List<VehicleType> ParseVehicleTypes(string path)
+        {
+            var csvStrings = CsvHelper.ReadCsvStrings(path);
+            foreach (var csvString in csvStrings)
+            {
+                VehicleTypes.Add(CreateVehicleType(csvString));
+            }
+            return VehicleTypes;
+        }
+
+        private List<Vehicle> ParseVehicles(string path)
+        {
+            var csvStrings = CsvHelper.ReadCsvStrings(path);
+            foreach (var csvString in csvStrings)
+            {
+                Vehicles.Add(CreateVehicle(csvString));
+            }
+            return Vehicles;
```

Comment:
It's pretty interesting decision but this code `Vehicles = ParseVehicles(vehiclesFileName)` looks like strange in this case. You already can use key word yield in this method. Just try. You will enjoy! :)

Reply:
fixed